### PR TITLE
1Password Connect Improved Networking

### PR DESCRIPTION
**Pull request by klaus385**
# 1Password Connect - (Improved Networking)

The changes I made here will allow for a multitude of things for users trying to interact with the 1Password Connect and Sync applications deployed.

## Changes

Below is a list of changes with detailed descriptions of the changes I made and why I made them.

### Documentation

1. Updated `values.yaml` to have the individual comments to allow for an inline view
2. Re-ordered `values.yaml` to have a flow that tries to follow a purpose for each section.

### Syntax

1. Removed trailing whitespace.
2. Added some templated for the improvements section that will allow ingress definitions and load balancer definitions.

### Explanation of Changes

1. Updated the `values.yaml` to have an example to use the `ingress.yaml` template, additional functionality for the `service.yaml` to use `loadBalancerIP` which would satisfy [this](https://github.com/1Password/connect-helm-charts/issues/88) GitHub issue, and `loadBalancerSourceRanges` to further extend on the service configuration for Load Balancer type.
2. Created the `ingress.yaml` template to allow for the ingress values to be deployed.
3. Created the `service.yaml` changes to allow for what is mentioned above.
4. Updated `_helpers.tpl` to allow for easy means of templating out `loadBalancerIP` and `loadBalancerSourceRanges`.

### Future Improvements 

This section is kinda a QA in regards to this helm chart. In which I had other changes in mind but decided to **NOT** at this time. This is related to how the `service.yaml` and `connect-deployment.yaml` is written.

#### Potential Future Changes Questions

1. Within `service.yaml` the API and Sync service are defined. Was there a particular reason why these were or are **NOT** separate templates? I ask this question due to the nature of needing to make a change to `service.yaml` in which by doing so you don't affect just API or Sync you affect change essentially to both.
2. Within `connect-deployment.yaml` similar question in regards to splitting these deployments out. Because, if let's say one wants to affect how API or Sync have deployed any changes to one effect the code which indirectly affects the other.
3. For `ingress.yaml` leverage if/else to discern the port when `.Values.connect.tls.enabled` is `true` or `false`.

### CODE REVIEW

**Comment by klaus385** [comment]
Just curious if there is anything else needed for this to get merged?

**Comment by klaus385** [comment]
@edif2008 maybe when I made this PR I missed something. Anyway, we can get some traction on this?

**Comment by klaus385** [comment]
Thought to bump this again and see what can be done to look into merging this into the helm chart?

**Comment by zifeo** [comment]
@florisvdg Is there anything blocking this to get reviewed/merged?

**Comment by edif2008** [comment]
Hey folks.
Thank you so much for your contribution in making the helm charts better. It's so cool to see this big enhancement made by the community.

I will apologise for keeping this PR hanging for a while. I've brought this PR on my team's radar to review it as soon as possible.

File: charts/connect/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+              serviceName: {{ $serviceName }}-https
+              servicePort: {{ $apiHttpsPort }}
+              {{- else }}
+              serviceName: {{ $serviceName }}
+              servicePort: {{ $apiHttpPort }}
+              {{- end }}
+              {{ end }}
+        {{- end }}
+  {{- end }}
+{{- end }}
```
Comment (volodymyrZotov) [inline charts/connect/templates/ingress.yaml:79]: Please add empty line at the end

File: charts/connect/values.yaml [hunk 2]
```diff
@@ -2,81 +2,243 @@
...
+  tls:
+    # Denotes whether the Connect API is secured with TLS
+    enabled: false
+    # The name of the secret containing the TLS key (tls.key) and certificate (tls.crt)
+    secret: op-connect-tls
+
+  # Ingress allows ingress services to be created to allow external access
+  # from Kubernetes to access 1Password Connect pods.
+  # In order to expose the service, use the route section below
+  ingress:
```
Comment (volodymyrZotov) [inline charts/connect/values.yaml:111]: Please add all new values you've added here to the README Values section.

File: charts/connect/values.yaml [hunk 3]
```diff
@@ -2,81 +2,243 @@
...
+  applicationName: onepassword-connect
+
+  # The type of Service resource to create for the Connect API and sync services.
+  # See: https://kubernetes.io/docs/concepts/services-networking/service
+  # This by default is NodePort and can also be defined as LoadBalancer.
+  # If serviceType is LoadBalancer then loadBalancerSourceRanges and loadBalancerIP should be defined.
+  # See: https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer
   serviceType: NodePort
+
+  #loadBalancerSourceRanges:
```
Comment (volodymyrZotov) [inline charts/connect/values.yaml:33]: Just not to forget that there is a lint error to fix

**Review by volodymyrZotov — CHANGES_REQUESTED** [review]

**Comment by volodymyrZotov** [comment]
@klaus385 Hey. I looked at the PR. Looks very good! Thank you! 👍  
 I left several small comments, please take a look.

File: charts/connect/templates/ingress.yaml (hunk 1 above)
Comment (klaus385) [reply charts/connect/templates/ingress.yaml:79]: Thanks for the feedback I will fix these later today :)

Comment (klaus385) [reply charts/connect/templates/ingress.yaml]: This has been updated :)

File: charts/connect/values.yaml (hunk 2 above)
Comment (klaus385) [reply charts/connect/values.yaml:111]: @volodymyrZotov i updated the `README.md` accordingly let me know if any more context needs to be added.

File: charts/connect/values.yaml (hunk 3 above)
Comment (klaus385) [reply charts/connect/values.yaml]: This should show green now when the CI is approved.

**Review by volodymyrZotov — APPROVED** [review]

**Comment by jpcoenen** [comment]
Thanks for your contribution, @klaus385! 🙌

**Pull request merged**